### PR TITLE
fix(debug): add predictable axis labels sorting order

### PR DESCRIPTION
**Pull request by markov00**
## Summary

This commit adds a predictable label sorting order for axis labels in the debug state.
The configured sorting order should always reflect the order starting from the origin of the axis, independently from the rotation or the axis position.
This is the only order that remains constant and predictable.

## Details

These changes are required to fix the functional tests in Kibana, providing a constant order to the axis labels.

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] Unit tests have been added or updated to match the most common scenarios

### CODE REVIEW

File: packages/charts/src/chart_types/xy_chart/state/selectors/get_debug_state.ts [hunk 1]
```diff
@@ -40,43 +44,63 @@ export const getDebugStateSelector = createCustomCachedSelector(
...
+      const sortingOrder = isHorizontalAxis(position)
+        ? rotation === 0 || rotation === 90
+          ? Predicate.NumAsc
+          : Predicate.NumDesc
+        : rotation === 0 || rotation === -90
+        ? Predicate.NumDesc
+        : Predicate.NumAsc;
+      const visibleTicks = [...geom.visibleTicks]
+        .sort(getPredicateFn(sortingOrder, 'position'))
+        .filter(({ label }) => label !== '');
```
Comment (monfera) [inline packages/charts/src/chart_types/xy_chart/state/selectors/get_debug_state.ts:88]: Let's filter first; in this case there.s no need for sorting something that gets removed, and no need for the `[...geom.visibleTicks]` shallow copy, as the filter creates a fresh array

**Review by monfera — APPROVED** [review]
LGTM! One minor optional code comment. Also, the sorting order is conditionally ascending or descending; it may be simpler for both our side and the recipient side to always do the same sorting, eg. `NumAsc`, because the `NumDesc` is arbitrary/artificial anyway. I'm not aware of a reason for complicating "business logic" for such a debug step, and the ascending order is a good standard, as noone cares about the physical screenspace order (imo)

File: packages/charts/src/chart_types/xy_chart/state/chart_state.debug.test.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+            id: 'left',
+            position: Position.Left,
+            integersOnly: true,
+          }),
+        ],
+        store,
+      );
+      const debugState = getDebugStateSelector(store.getState());
+      expect(debugState.axes?.x[0].labels).toEqual(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10']);
+      expect(debugState.axes?.y[0].labels).toEqual(['0', '1', '2', '3', '4', '5', '6', '7', '8', '9', '10']);
```
Comment (nickofthyme) [inline packages/charts/src/chart_types/xy_chart/state/chart_state.debug.test.ts:49]: How about `toMatchSnapshot`?

```suggestion
      expect(debugState).toMatchSnapshot();
```

**Comment by markov00** [comment]
> LGTM! One minor optional code comment. Also, the sorting order is conditionally ascending or descending; it may be simpler for both our side and the recipient side to always do the same sorting, eg. `NumAsc`, because the `NumDesc` is arbitrary/artificial anyway. I'm not aware of a reason for complicating the order with "business logic" for such a debug step, and the ascending order is a good standard, as noone cares about the physical screenspace order (imo)

You are right, but I thought it was easier to grasp if we connect that to a "bottom-left" screen space. When we are going to reverse the way we compute the Y position on cartesian we can clean up this code (currently it follows the top-left 0,0 origin, that is actually harder to reason about that a bottom-left one)

File: packages/charts/src/chart_types/xy_chart/state/chart_state.debug.test.ts (hunk 2 above)
Comment (nickofthyme) [reply packages/charts/src/chart_types/xy_chart/state/chart_state.debug.test.ts:49]: This would cover other unexpected changes to debug states.

**Review by nickofthyme — APPROVED** [review]
LGTM

File: packages/charts/src/chart_types/xy_chart/state/chart_state.debug.test.ts (hunk 2 above)
Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/state/chart_state.debug.test.ts]: done ba38aa9

File: packages/charts/src/chart_types/xy_chart/state/selectors/get_debug_state.ts (hunk 1 above)
Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/state/selectors/get_debug_state.ts]: you are right, my bad, done in c407bba

**Pull request merged**